poetro_app/lib/features/search/widgets/poetry_list_view.dart

```diff
@@ -0,0 +1,88 @@
+import 'dart:developer';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:infinite_scroll_pagination/infinite_scroll_pagination.dart';
+import 'package:poetro_app/features/poetry/presentation/bloc/poetry_bloc/poetry_bloc.dart';
+import 'package:poetro_app/features/poetry/presentation/models/poetry_model.dart';
+import 'package:poetro_app/features/poetry/presentation/widgets/poetry_previw_item.dart';
+
+class PoetryListView extends StatefulWidget {
+  final String value;
+  const PoetryListView({super.key, required this.value});
+
+  @override
+  createState() => _PoetryListViewState();
+}
+
+class _PoetryListViewState extends State<PoetryListView> {
+  late PagingController<int, PoetryModel> pagingController;
+
+  @override
+  void initState() {
+    pagingController = PagingController(firstPageKey: 1);
+    pagingController.addPageRequestListener(fetchData);
+    super.initState();
+  }
+
+  Future<void> fetchData(int pageKey) async {
+    // log("Fetching data for page: $pageKey", name: 'PoetryListView');
+    log("Value: ${widget.value}", name: 'PoetryListView');
+
+    widget.value.isEmpty
+        ? context.read<PoetryBloc>().add(
+              PoetryEvent.fetchRandomSequencePoems(2),
+            )
+        : context.read<PoetryBloc>().add(
+              PoetryEvent.fetchPoetryByTitle(widget.value, 2),
+            );
+  }
```

Comment:
why this logic here if you arleady fecthed poetry in searcg page
and you named widget poetry listview but in initstate you are fetching somthing